include/zephyr/llext/symbol.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_LLEXT_SYMBOL_H
+#define ZEPHYR_LLEXT_SYMBOL_H
+
+#include <zephyr/sys/iterable_sections.h>
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+/**
+ * @brief Linkable loadable extension symbol
+ * @defgroup llext_symbols LLEXT symbols
+ * @ingroup llext
+ * @{
+ */
+
+/**
+ * @brief A symbol (named memory address)
+ */
+struct llext_symbol {
+	/* string name of the symbol */
+	char *name;
+
+	/* If an object is exported, it might be writable, so no "const" */
```

Comment:
if this is API, this also needs an actual doxygen comment describing what the `addr` field is for.